### DISCUSSION TITLE
Prevent Google from crawling/indexing chunks and json files

### DISCUSSION THREAD

**Discussion by baj84** [vercel/next.js#34193] [Help]
Hi,

I noticed Google is crawling and attempting to index my nextjs chunk and JSON files.
e.g.
https://www.mysite.com/_next/data/wCiKzgJzU2VNoiVjrCRBB/login.json
https://www.mysite.com/_next/static/chunks/pages/_app-459fc7accb5648d12ee2.js
...

Is there a way to prevent this?

Thanks

**Comment by stevemarksd** [comment]
Add a header `X-Robots-Tag` with value `noindex` on those files. https://developers.google.com/search/docs/advanced/robots/robots_meta_tag

Or in your robots ignore `/_next/*`

**Reply by gettoabhi88** [reply]
But if we block next files this can deindex the website. Recently, this happened with me. I tried blocking next_data and next_static files which hampered the website index and now I am only left with 13 pages indexed on google.

**Comment by GMaiolo** [comment]
@baj84 Did you ever find a solution for this or it didn't matter at the end?

**Comment by FloPiantanida** [comment]
Hello,
We get the same issue, Google is crawling _next directory.
The result is than we get a lot of error 500 on old chunk files.
We can't add it in robots beacuse the page will not be load correctly (bad lighthouse score)
Do you have any solution to prevent error 500 in google search console on old chunk?

**Comment by Branch** [comment]
I'm also experiencing this "issue". Is it OK from an SEO perspective to exclude the _next folder or not?

**Comment by KrustyC** [comment]
Also experiencing the same problem here, is there any workaround?

**Comment by vforvalerio87** [comment]
What I'd suggest:

- In NGINX (or whatever web server you have in front of your next application) pattern-match calls to .json endpoints
- Check the "Sec-Fetch-Mode" header (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Mode)
- If it's "navigate", return 403 and add the response header "X-Robots-Tag: noindex" (https://developers.google.com/search/docs/crawling-indexing/robots-meta-tag)
